c7n/resources/sfn.py

```diff
@@ -32,3 +33,57 @@ class resource_type(object):
             "describe_state_machine", "stateMachineArn",
             'stateMachineArn', None)
         filter_name = None
+
+
+@StepFunction.action_registry.register('tag')
+class TagStepFunction(Tag):
+    """Action to create tag(s) on a step function
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-step-function
+                resource: step-machine
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-tag-value
+    """
+
+    permissions = ('stepfunctions:TagResource',)
+
+    def process_resource_set(self, client, resources, tags):
+
+        tags_lower = []
+
+        for tag in tags:
+            tags_lower.append({k.lower(): v for k, v in tag.items()})
```

Comment:
why is this mutating the keys the user configured?

Reply:
So, oddly enough, the boto3 requests around step functions have to be lower case or they fail.  It's strangely inconsistent with other boto3 commands.  

StepFunctions:
`response = client.tag_resource(
    resourceArn='string',
    tags=[
        {
            'key': 'string',
            'value': 'string'
        },
    ]
)`

FSx:
`response = client.tag_resource(
    ResourceARN='string',
    Tags=[
        {
            'Key': 'string',
            'Value': 'string'
        },
    ]
)`

Reply:
oh this is on the key dictionary not the key itself, i see, that makes sense thanks. yeah tagging apis are widely inconsistent, where possible we try to use the resource group tagging api as it supports bulk ops and provides a consistent api.